Obsolete forward declarations validParams<Material>

## Bug Description
There is one forward declaration `validParams<Material>` in `Material.h` that looks obsolete and results in a linking error:
https://github.com/idaholab/moose/blob/f9a2a7b396f3d5313c5b58e36362963a8347d88f/framework/include/materials/Material.h#L33-L34

## Steps to Reproduce
I tried to compile my MOOSE application with gcc 11.2.0 on our cluster and got the linker error
```
undefined reference to InputParameters validParams<Material>()
```

## Impact
The error prevents from compiling my code (though I can easily remove the forward declarations and everything gets fine).

--- a/framework/include/materials/Material.h
+++ b/framework/include/materials/Material.h
@@ -27,12 +27,6 @@
   using Material::_current_subdomain_id;                                                           \
   using Material::_current_side
 
-// forward declarations
-class Material;
-
-template <>
-InputParameters validParams<Material>();
-
 /**
  * Materials compute MaterialProperties.
  */

